src/Symfony/Component/Console/Tests/Command/CommandTest.php

```diff
@@ -348,6 +349,57 @@ public function callableMethodCommand(InputInterface $input, OutputInterface $ou
     {
         $output->writeln('from the code...');
     }
+
+    public function testLockIsReleased()
+    {
+        if (!class_exists('Symfony\Component\Filesystem\LockHandler')) {
+            $this->markTestSkipped();
+        }
+
+        $command1 = new \FooLock1Command();
+
+        $tester = new CommandTester($command1);
+        $tester->execute(array());
+        $tester->execute(array());
+    }
+
+    /**
+     * @expectedException Symfony\Component\Console\Exception\RuntimeException
+     * @expectedExceptionMessage <error>The command "foo:lock1" is already running in another process.</error>
+     */
+    public function testLockThrowsExceptionIfAlreadyLocked()
+    {
+        if (!class_exists('Symfony\Component\Filesystem\LockHandler')) {
+            $this->markTestSkipped();
+        }
+
+        $command1 = new \FooLock1Command();
+
+        $lock = new LockHandler($command1->getName());
+        $lock->lock();
+
+        $tester = new CommandTester($command1);
+        $tester->execute(array());
+
+        $lock->release();
+    }
+
+    public function testLockSilentModeDoesNotThrowAnyException()
+    {
+        if (!class_exists('Symfony\Component\Filesystem\LockHandler')) {
+            $this->markTestSkipped();
+        }
+
+        $command1 = new \FooLock2Command();
+
+        $lock = new LockHandler($command1->getName());
+        $lock->lock();
+
+        $tester = new CommandTester($command1);
+        $tester->execute(array());
+
+        $lock->release();
```

Comment:
This does not really test a good behavior . You could for example return in the command the return of the lock handler, and check here the value.